exporter/exporter.py

```diff
@@ -0,0 +1,168 @@
+    )
+
+
+# T4C determines the project name based on the directory name
+def get_model_dir_with_project_name(path: pathlib.Path) -> pathlib.Path:
+    project_dir: pathlib.Path = pathlib.Path(f"/tmp/{T4C_PROJECT_NAME}")
+    project_dir.mkdir(parents=True)
+
+    for _file in path.glob("*"):
+        shutil.copy2(_file, project_dir)
+
+    return project_dir
+
+
+def run_exporter_script(model_dir: pathlib.Path) -> None:
+    t4c_repo_name = os.environ["T4C_REPO_NAME"]
+
+    command: list[str] = [
+        "/opt/capella/capella",
+        "-consoleLog",
+        "-data",
+        "workspace",
+        "-application",
+        "com.thalesgroup.mde.melody.collab.exporter",
+        "-closeserverOnFailure",
+        "false",
+        "-overrideExistingProject",
+        "true",
+        "-mergeDifferenceOnExistingProjects",
+        "true",
+        "-hostname",
+        os.environ["T4C_REPO_HOST"],
+        "-port",
+        os.getenv("T4C_REPO_PORT", "2036"),
+        "-repoName",
+        t4c_repo_name,
+        "-repositoryLogin",
+        os.environ["T4C_USERNAME"],
+        "-repositoryPassword",
+        os.environ["T4C_PASSWORD"],
+        "-httpLogin",
+        os.environ["HTTP_LOGIN"],
+        "-httpPassword",
+        os.environ["HTTP_PASSWORD"],
+        "-httpPort",
+        os.environ["HTTP_PORT"],
+        "-sourceToExport",
+        str(model_dir),
+    ]
+
+    with subprocess.Popen(
+        command, cwd="/opt/capella", stdout=subprocess.PIPE, text=True
+    ) as popen:
+        if popen.stdout:
+            for line in popen.stdout:
+                print(line, end="")
+                if "Team for Capella server unreachable" in line:
+                    raise RuntimeError(
+                        f"{ERROR_PREFIX} - Team for Capella server unreachable"
+                    )
+                elif "Repository not found" in line:
+                    raise RuntimeError(
+                        f'{ERROR_PREFIX} - Repository "{t4c_repo_name}" does not exist'
+                    )
+                elif "No address associated with hostname" in line:
+                    raise RuntimeError(f"{ERROR_PREFIX} - Unknown host")
+
+    if (return_code := popen.returncode) != 0:
+        raise subprocess.CalledProcessError(return_code, command)
+
+    log.info("Export of model to TeamForCapella server finished")
+
+
+def is_capella_5_x_x() -> bool:
+    return bool(re.match(r"5.[0-9]+.[0-9]+", os.getenv("CAPELLA_VERSION", "")))
+
+
+if __name__ == "__main__":
+    check_capella_version()
+
+    file_handler = os.getenv("FILE_HANDLER", "")
+    if file_handler == "local":
+        _model_dir = pathlib.Path("/tmp/data")
+    else:  # USE GIT
+        _git_dir = checkout_git_repository()
+        _model_dir = determine_model_dir(_git_dir)
+
+    check_dir_for_aird_file(_model_dir)
+    _project_dir: pathlib.Path = get_model_dir_with_project_name(_model_dir)
+
+    run_exporter_script(_project_dir)
+
+    print("Export finished")
```

Comment:
Personally, I would keep it like this or look at this when my tests are fixed because I am not sure really sure if the `log.info` is actually printed (I have never paid attention to whether both lines come directly above each other at the end). And if thats the case, I think it would be better to keep the line at the end of the `__main__`  since this makes it clear that this should be the last line (otherwise someone might add things after the `run_exporter_script` in the future)

Reply:
If `log.info` doesn't work, we should make it work :) 
In case we'd keep the `Export finished` line, I'd change it to a logging call instead.

Reply:
Yes, we can adapt this, but there are 1 1/2 problems from my side. The 1/2 problem is that I am not 100% sure how `print` and `log` behave here. This is because the "Export finished" line appears before most of the other lines are printed (so in the current state it is not really the last line). Obviously this is not our intended behaviour, so setting `flush=True` for the `print` of the sub-process logs seems to fix this problem (but since I am not 100% sure, I can't say that it always works well).
The other issue I have with this approach is that we use `log.info` for the success message but what happens when we set the log level to `WARNING`. So if we are fine that in this case no success message is shown then we can change it but should then also adjust it for our `backups.py`.

Reply:
> The other issue I have with this approach is that we use `log.info` for the success message but what happens when we set the log level to `WARNING`. So if we are fine that in this case no success message is shown then we can change it but should then also adjust it for our `backups.py`.

When the user doesn't want to have INFO messages, the user doesn't get INFO messages :) 
I'd say "Export finished" is also just an informal message.